src/Query.php

```diff
@@ -558,11 +558,11 @@ protected function _execute()
     {
         $connection = $this->_repository->getConnection();
         $index = $this->_repository->getName();
-        $type = $connection->getIndex($index)->getType($this->_repository->getType());
+        $searchObject = $connection->getIndex($index)->createSearch($this->_repository->getType());
```

Comment:
The current syntax of [`createSearch()`](https://elastica.io/api/latest/classes/Elastica.Index.html#method_createSearch) expect a `string|array|\Elastica\Query` as first parameter.
Maybe we can replace the instantiation of `$searchObject` with directly use [`Index::search`](https://elastica.io/api/latest/classes/Elastica.Index.html#method_search), just like:

```php
return $connection->getIndex($index)->search($query, $this->_searchOptions);
```

Reply:
Hi, good point. I've changed it now. Hope it works